### PR TITLE
Change the formula for monster strength calculation

### PR DESCRIPTION
**Pull request by Ben80-hero**
Fix a game analytics bug in Monster::GetMonsterStrength.

Here in this line:
const double attackDefense = 1.0 + attack * 0.1 + defense * 0.05;

More correctly like this:
const double attackDefense = 1.0 + attack * 0.1 + defense * 0.1;

Rationale:

1) In practice, damage in battle is determined by the difference between attack parameter (of the 1st monster) and defense parameter (2nd monster), and not by some formula that would include parameters only one of the monsters. Therefore, in practice, as a result, one unit of defense parameter mutually
annihilates with one unit of the opponent's attack parameter. In this sense, the value attack and defense are equivalent.
Example:
Consider a battle (exchange of blows) of 2 hypothetical monsters. The 1st monster has attack parameter is 15 and defense parameter is 5.
 2nd monster has attack is 5, and the defense is 15. In practice, these 2 monsters in the battle with each other are equal. However according to the formula above, for the 1st monster the function will return 2.75, and for the 2nd monster it will return 2.25.

2) Consider the "abstract" value of these 2 parameters (for example, when considering combat against a hypothetical monster with an attack equal 0 and a defense equal 0).

It turns out that the effect of increasing attack parameter increases linearly (we consider the damage dealt by this monster):
If the monster has attack equal 5, then the ratio of total damage to base damage is 1.5.
If the monster has attack equal 10, then the ratio of total damage to base damage is 2.0.
If the monster has attack equal 15, then the ratio of total damage to base damage is 2.5.

The effect of the growth of the defense parameter grows non-linearly (we consider the damage dealt to this monster):
If the monster has defense equal of 5, then the ratio of total damage to base damage is 1.33.
If the monster has defense equal of 10, then the ratio of total damage to base damage is 2.0.
If the monster has defense equal of 15, then the ratio of total damage to base damage is 4.0.
(however, in the original game there is a limiter, so in practice ratio 4.0 will not be reached)

P.S. It was this logic that NWC game analysts adhered to. In both Heroes 2 and Heroes 3 they consider the value of attack and defense parameters as equal.

### CODE REVIEW

File: src/fheroes2/monster/monster.cpp [hunk 1]
```diff
@@ -179,7 +179,7 @@ double Monster::GetMonsterStrength( int attack, int defense ) const
     if ( defense == -1 )
         defense = GetDefense();
 
-    const double attackDefense = 1.0 + attack * 0.1 + defense * 0.05;
+    const double attackDefense = 1.0 + attack * 0.1 + defense * 0.1;
```
Comment (ihhub) [inline src/fheroes2/monster/monster.cpp:182]: Hi @Ben80-hero , please optimize the formula as we have 2 identical multipliers.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @Ben80-hero , I left one optimization comment and also how did you test the behavior of AI for this change? Has it become better? If yes for which cases?

File: src/fheroes2/monster/monster.cpp (hunk 1 above)
Comment (Ben80-hero) [reply src/fheroes2/monster/monster.cpp:182]: I did not conduct testing, and I doubt that tests are possible or necessary in this case. That is, in this case, we simply refine one of the estimates on which the actions of the AI ​​are based. The more accurate the AI's estimates, the better its overall decisions.

I think that these changes will have the greatest effect on high-level creatures - their power was clearly underestimated.

**Comment by oleg-derevenetz** [comment]
This formula (with exactly the same coefficients) was initially added by @idshibanov in #692 and it probably has some grounds. Let's ask him for a review.

File: src/fheroes2/monster/monster.cpp (hunk 1 above)
Comment (Ben80-hero) [reply src/fheroes2/monster/monster.cpp]: Have optimized formula.

**Pull request closed without merging**